exonum-java-binding/core/src/main/java/com/exonum/binding/core/runtime/TransactionMethodExtractor.java

```diff
@@ -84,27 +87,42 @@ private static void checkDuplicates(Map<Integer, Method> transactionMethods, int
    */
   private static void validateTransactionMethod(Method transaction, Class<?> serviceClass) {
     String errorMessage = String.format("Method %s in a service class %s annotated with"
-        + " @TransactionMethod should have precisely two parameters of the following types:"
-        + " 'byte[]' and 'com.exonum.binding.core.transaction.TransactionContext'",
+        + " @TransactionMethod should have precisely two parameters: transaction arguments of"
+        + " 'byte[]' type or a protobuf type and transaction context of"
+        + " 'com.exonum.binding.core.transaction.TransactionContext' type",
         transaction.getName(), serviceClass.getName());
     checkArgument(transaction.getParameterCount() == 2, errorMessage);
     Class<?> firstParameter = transaction.getParameterTypes()[0];
     Class<?> secondParameter = transaction.getParameterTypes()[1];
-    checkArgument(firstParameter == byte[].class,
+    checkArgument(firstParameter == byte[].class || isProtobufArgument(firstParameter),
         String.format(errorMessage
             + ". But first parameter type was: %s", firstParameter.getName()));
     checkArgument(TransactionContext.class.isAssignableFrom(secondParameter),
         String.format(errorMessage
             + ". But second parameter type was: %s", secondParameter.getName()));
   }
 
-  private static MethodHandle toMethodHandle(Method method, Lookup lookup) {
+  private static TransactionMethodObject toTransactionMethodObject(Method method, Lookup lookup) {
+    Serializer argumentsSerializer = null;
```

Comment:
I suggest specifying explicitly that the Serializer is unbounded <?> — in our case,
    may produce any Object in its `fromBytes` (in TransactionMethodObject API and in the variables declarations)
    so that the intent is clear and there are no warnings.